### DISCUSSION TITLE
Is there any porting guide avaliable?

### DISCUSSION THREAD

**Discussion by lida2003** [Firmament-Autopilot/FMT-Firmware#118] [Ideas]
Hi

I'm New to FMT. 

I had a general view of the project, and I have a [Holybro Kakute F7 Flight Controller AIO](https://hobbyking.com/en_us/kakute-f7-aio.html?___store=en_us).

Is there any way to make this fc work with FMT?

**Comment by JcZou** [comment]
Hi
Thanks for your interest. Yes, it can be ported to support F7 as FMT already support F7 with CUAV V5+ and Pixhawk 4 BSP.
You can refer to [document](https://firmament-autopilot.github.io/FMT-DOCS/#/developer/drivers) to know how to port driver.
And there is a minimum system that you can refer to: https://github.com/Firmament-Autopilot/FMT-Firmware/tree/master/target/minimum/cuav/v5_plus

**Reply by lida2003** [reply]
Links above provided, it's kind of how to write a driver not the target board porting guide, especially F7/H7 series. Is there any step by step guides, just like [APM - Porting to a new flight controller board](https://ardupilot.org/dev/docs/porting.html)?
